src/python/pants/engine/legacy/graph.py

```diff
@@ -519,6 +544,29 @@ def owns_any_source(legacy_target: HydratedTarget) -> bool:
   return TransitiveHydratedTargets(tuple(tht.root for tht in transitive_hydrated_targets), closure)
 
 
+@rule
+async def legacy_transitive_hydrated_targets(
```

Comment:
Should make sure we don't see a performance impact on legacy goals for this... things like `dependencies`, maybe.

Reply:
How can we do that? Could we possibly land this for the dev release and use Twitter's performance suite to check if there were any regressions in the dev release?

Reply:
I would love to see an upstream performance suite, personally...........